optuna/distributions.py

```diff
@@ -190,27 +200,21 @@ class UniformDistribution(BaseDistribution):
     """
 
     def __init__(self, low: float, high: float) -> None:
+        super().__init__(low=low, high=high, log=False, step=None)
 
-        if low > high:
-            raise ValueError(
-                "The `low` value must be smaller than or equal to the `high` value "
-                "(low={}, high={}).".format(low, high)
-            )
-
-        self.low = float(low)
-        self.high = float(high)
-
-    def single(self) -> bool:
-
-        return self.low == self.high
-
-    def _contains(self, param_value_in_internal_repr: float) -> bool:
+    def _asdict(self) -> Dict:
+        d = copy.deepcopy(self.__dict__)
+        d.pop("log")
+        d.pop("step")
+        return d
 
-        value = param_value_in_internal_repr
-        return self.low <= value <= self.high
+    def __repr__(self) -> str:
+        kwargs = ", ".join("{}={}".format(k, v) for k, v in sorted(self._asdict().items()))
+        return "{}({})".format(self.__class__.__name__, kwargs)
```

Comment:
I'm not sure but if we change the `BaseDistribution`'s `__repr__` to use `self._asdict`, would it not be necessary to implement `__repr__` for `UniformDistribution`, etc.?

Reply:
It makes sense to me. I didn't do so because I focus only updating `__repr__` to fix failed tests but it would make codebase more simple. I revised the PR in 970d1c0.

@hvy 
Please give us a comment if you have. I mention you because you also have edited `__repr__` for `IntLogUniformDistribution`.
https://github.com/optuna/optuna/blob/bb0ad9a77e2c7715a7bddb82cc9db8ac321a39c7/optuna/distributions.py#L505-L509

Reply:
Sorry, I cannot remember the exact details but I think your change in encapsulating the "formatting" in `_asdict` is what I meant. Comment was introduced in https://github.com/optuna/optuna/pull/1438. Just for the record, `_asdict` has an even longer history, https://github.com/optuna/optuna/pull/16.